app/models/levelling_up_premium_payments/eligibility.rb

```diff
@@ -57,10 +57,39 @@ def policy
 
     # maintains interface
     def ineligible?
+      if current_school.present?
+        !LevellingUpPremiumPayments::SchoolEligibility.new(current_school).eligible?
```

Comment:
Isn't this just
```ruby
current_school.present? && !LevellingUpPremiumPayments::SchoolEligibility.new(current_school).eligible?
```

You can move this to a method.

Reply:
Done